lib/autoproj/cli/main_github.rb

```diff
@@ -94,13 +95,21 @@ def update_package_set(url)
                 end
 
                 def perform_update
-                    Update.new(ws).run(
-                        [], autoproj: false,
-                            packages: false,
-                            config: true,
-                            deps: false,
-                            osdeps: false
-                    )
+                    begin
+                        Update.new(ws).run(
+                            [], autoproj: false,
+                                packages: false,
+                                config: true,
+                                deps: false,
+                                osdeps: false
+                        )
+                    rescue Autoproj::PackageNotFound
```

Comment:
I'm not entirely sure about this ... Do you really want to leave the user's workspace in an unusable state ?

Reply:
This is a step in the update cycle. Hopefully, once the user updates the workspace everything will be ok. If theyŕe not, then that is a problem in the PR he/she is testing. I feel like having a "broken" workspace is one of the possible outcomes of a PR that removes a package from a package set.

What do you suggest?

Reply:
Sorry, I still don't get it.

`perform_update` is called [before the overrides are generated](https://github.com/rock-core/autoproj-github/pull/2/files/b04644cb9f2a98237ffee65a49270525d1d85547#diff-3cfc24b05f4b19ce7cf60a3ecc3e3192L35), so what's the intention ?

Reply:
That's because there's also the case of testing a PR that adds a package that is not on master yet. So, to be able to resolve the PR url to package name the plugin needs to update and load the package set PR that defines the package. We don't expect a PackageNotFound error in this case but, if you are also removing a package, unless you update the whole workspace, PackageNotFound errors are expected because other packages might depend on the one that is being removed. It is up to the user to do an `aup` and check if everything goes well (the plugin is not  testing the PR, it's only generating the overrides that allows the user to test it)

Reply:
OK, now I get why you might have an error (and I'm OK with that), but I still don't see how the code matches what you are describing. How I am reading the method, you do

1. update
2. generate overrides

So, when you run the update, the overrides are not yet applied ...

Reply:
The flow is:

1. The user runs `autoproj github overrides URL`;
2. The plugin fetches overrides for the packages it currently knows. If there's a PR to a package set we assume there may be missing entries (for packages that are not defined yet), so it restarts with `--update`;
3. The package sets are updated
4. New set of overrides is generated, now with more packages defined

Reply:
(2) is actually repeated until all overrides entries to package sets match the current VCSDefinition

Reply:
Aaaaaah. **Now** I get it :P

Reply:
Thanks for taking the time :+1: